packages/hub/writers.js

```diff
@@ -265,6 +276,7 @@ class PendingChange {
     this.serverProvidedValues = new Map();
     this._finalizer = finalizer;
     this._aborter = aborter;
+    let { included } = opts;
```

Comment:
I don't understand why included is getting special handling here. We pass whole jsonapi documents to writers and they pass whole documents back, so both cases necessarily already support included. Writers are free to not do anything with the included we give them, and they can choose to pass back anything they want to, and in either case the document context should still be able to do the right thing -- if present, we can use the included models directly without needing to load them, if missing and wanted, we will load them.

Reply:
the issue is like what I mentioned above, the writers are only saving the `data` portion of the document. they are actually ignoring the `included` part of the document. I can update the existing writers to actually care about the `included` part of the document, if that's the direction you'd like to go.

Reply:
Or maybe the write approach is to invoke the writers for each resource in the included part of the card document (so effectively this means that we write out the internal models of the document as their own rows in the index, which I think we've gone back and forth with)...